src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -183,20 +183,28 @@ public function getCookieJar()
     /**
      * Returns the current Crawler instance.
      *
-     * @return Crawler|null A Crawler instance
+     * @return Crawler A Crawler instance
      */
     public function getCrawler()
     {
+        if (null === $this->crawler) {
+            @trigger_error(sprintf('Calling the "%s()" method before the "request()" one is deprecated since Symfony 4.1 and will throw an exception in 5.0.', __METHOD__), E_USER_DEPRECATED);
```

Comment:
Can you add the exception that would be thrown in 5.0 and comment it? That will help when we remove the deprecation notices. Thanks.

Reply:
@fabpot Please review

Reply:
Sorry, I was not clear enough. We cannot add the `@throws` annotation yet as that would be wrong, but to ease remving the deprecation notices, you can add a line like this bellow the trigger call:

```
// throw new BadMethodCallException('...');
```

Reply:
Fixed